packages/core/src/loaders/geometryLoader.ts

```diff
@@ -6,31 +6,156 @@ import type {
+
+  // Broadcast a geometry loaded event once the geometry is loaded
+  geometryLoadObject.promise.then(
+    function (geometry) {
+      triggerEvent(eventTarget, Events.GEOMETRY_LOADED, { geometry });
+    },
+    function (error) {
+      const errorObject = {
+        geometryId,
+        error,
+      };
+
+      triggerEvent(eventTarget, Events.GEOMETRY_LOADED_FAILED, errorObject);
+    }
+  );
+
+  return geometryLoadObject;
+}
 
 /**
- * It creates a geometry object and caches it
- * @param geometryId - A unique identifier for the geometry.
- * @param options - GeometryOptions
- * @returns A promise that resolves to a geometry object.
+ * Loads a geometry given a geometryId and optional options and returns a promise which will resolve to
+ * the loaded geometry object or fail if an error occurred. The loaded geometry is not stored in the cache.
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader
+ *
+ * @returns An Object which can be used to act after a geometry is loaded or loading fails
  */
-async function createAndCacheGeometry(
+export function loadGeometry(
   geometryId: string,
-  options: GeometryOptions
+  options?: GeometryOptions
+): Promise<IGeometry> {
+  if (geometryId === undefined) {
+    throw new Error('loadGeometry: parameter geometryId must not be undefined');
+  }
+
+  let geometryLoadObject = cache.getGeometryLoadObject(geometryId);
+
+  if (geometryLoadObject !== undefined) {
+    return geometryLoadObject.promise;
+  }
+
+  geometryLoadObject = loadGeometryFromGeometryLoader(geometryId, options);
+
+  return geometryLoadObject.promise;
+}
+
+/**
+ * Loads a geometry given a geometryId and optional options and returns a promise which will resolve to
+ * the loaded geometry object or fail if an error occurred. The geometry is stored in the cache.
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader
+ *
+ * @returns Geometry Loader Object
+ */
+export async function createAndCacheGeometry(
+  geometryId: string,
+  options?: GeometryOptions
 ): Promise<IGeometry> {
+  if (geometryId === undefined) {
+    throw new Error(
+      'createAndCacheGeometry: parameter geometryId must not be undefined'
+    );
+  }
+
+  let geometryLoadObject = cache.getGeometryLoadObject(geometryId);
+
+  if (geometryLoadObject !== undefined) {
+    return geometryLoadObject.promise;
+  }
+
+  geometryLoadObject = loadGeometryFromGeometryLoader(geometryId, options);
+
+  await cache.putGeometryLoadObject(geometryId, geometryLoadObject);
+
+  return geometryLoadObject.promise;
+}
+
+/**
+ * Creates and caches a geometry synchronously
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader
+ *
+ * @returns IGeometry
+ */
+export function createAndCacheLocalGeometry(
```

Comment:
This one makes sense to call it createAndCacheGeometry, if the previous one is loadAndCacheGeometry then it makes more sense.